### PR TITLE
[GEMM] Add autotune configs with different prefetch distances

### PR DESCRIPTION
**Pull request by whitneywhtsang**
CI: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10761214762

- `4, 4096, 12288` improves 9.35% out of box.
- `8192, 16384, 1024` improves 2.93% on advanced path.
- other shapes improves up to ~1%.

Added 3 new configs (B=1):
```
        triton.Config(
            {'BLOCK_SIZE_M': 256, 'BLOCK_SIZE_N': 256, 'BLOCK_SIZE_K': 32, 'GROUP_SIZE_M': 4, 'grf_mode': 'large'},
            num_stages=1, num_warps=32),
        triton.Config(
            {'BLOCK_SIZE_M': 256, 'BLOCK_SIZE_N': 128, 'BLOCK_SIZE_K': 32, 'GROUP_SIZE_M': 4, 'grf_mode': 'large'},
            num_stages=3, num_warps=32),
        triton.Config(
            {'BLOCK_SIZE_M': 8, 'BLOCK_SIZE_N': 512, 'BLOCK_SIZE_K': 64, 'GROUP_SIZE_M': 1, 'grf_mode': 'large'},
            num_stages=3, num_warps=32)
```

### CODE REVIEW

File: benchmarks/triton_kernels_benchmark/gemm_benchmark.py [hunk 1]
```diff
@@ -205,26 +206,28 @@ def matmul(a, b):
...
+            [1, 16384, 1024, 8192],  #
+            [1, 16384, 4096, 8192],  #
+            [1, 16384, 8192, 1024],  #
+            [1, 16384, 8192, 4096],  #
+            [4, 32768, 128, 4096],  #
+            [4, 32768, 4096, 128],  #
+            [32, 4096, 4096, 128],  #
+            [4096, 8, 128, 16384],  #
+            [4096, 8, 16384, 128]
+        ],
```
Comment (whitneywhtsang) [inline benchmarks/triton_kernels_benchmark/gemm_benchmark.py:230]: NFC

File: benchmarks/triton_kernels_benchmark/gemm_benchmark.py [hunk 2]
```diff
@@ -86,22 +86,23 @@ def matmul_kernel_with_block_pointers(
...
+    ] + [
         triton.Config(
             {'BLOCK_SIZE_M': 8, 'BLOCK_SIZE_N': 512, 'BLOCK_SIZE_K': 64, 'GROUP_SIZE_M': 1, 'grf_mode': 'large'},
-            num_stages=2, num_warps=32),
+            num_stages=s, num_warps=32) for s in [2]
+    ] + [
         triton.Config(
             {'BLOCK_SIZE_M': 8, 'BLOCK_SIZE_N': 128, 'BLOCK_SIZE_K': 64, 'GROUP_SIZE_M': 1, 'grf_mode': 'large'},
-            num_stages=2, num_warps=4),
+            num_stages=s, num_warps=4) for s in [2]
```
Comment (whitneywhtsang) [inline benchmarks/triton_kernels_benchmark/gemm_benchmark.py:105]: NFC

File: benchmarks/triton_kernels_benchmark/gemm_benchmark.py [hunk 3]
```diff
@@ -86,22 +86,23 @@ def matmul_kernel_with_block_pointers(
...
+    ] + [
         triton.Config(
             {'BLOCK_SIZE_M': 64, 'BLOCK_SIZE_N': 128, 'BLOCK_SIZE_K': 32, 'GROUP_SIZE_M': 4, 'grf_mode': 'large'},
-            num_stages=2, num_warps=32),
+            num_stages=s, num_warps=32) for s in [2]
+    ] + [
         triton.Config(
             {'BLOCK_SIZE_M': 8, 'BLOCK_SIZE_N': 512, 'BLOCK_SIZE_K': 64, 'GROUP_SIZE_M': 1, 'grf_mode': 'large'},
-            num_stages=2, num_warps=32),
+            num_stages=s, num_warps=32) for s in [2]
```
Comment (LiyangLingIntel) [inline benchmarks/triton_kernels_benchmark/gemm_benchmark.py:101]: If we do not want more num_stages options, do we still need to use it here?

**Review by LiyangLingIntel — APPROVED** [review]
I think it's good to me for now before we have more insights on optimal prefetch distance options.

**Pull request merged**